engine/src/sellst.cpp

```diff
@@ -539,90 +499,101 @@ Boolean MCSellist::cut()
 
 Boolean MCSellist::del()
 {
-    if (!IsDeletable())
-        return False;
- 
-	if (nullptr == objects)
+    if (!IsDeletable() || m_objects.empty())
         return False;
 
     MCundos->freestate();
-    
-    MCStack *sptr = objects->m_ref->getstack();
-    while (objects != NULL)
+
+    MCStack *sptr = m_objects.front()->getstack();
+    /* When we delete an object, it removes itself from the selection,
```

Comment:
Is this actually the case in the existing code? It looks like it removes the selnode from the list (calling objects->remove() - which is not MCSellist::remove) and then calls 'del' - removal of the selnode from the list will mean that it won't be in the selection when del() is called, so the trying to remove the object from the sellist (in del()) won't do anything - in particular, send a message).

I think I see how the O(n^2) behavior might have existed before, but now that the is_selected flag is tied to the lifetime of a Selnode in which the objecthandle sits, when the selnode is removed, the object will no longer be selected so shouldn't touch sellist at all in del() (assuming calling MCsellist->remove(..) in del() is predicated on the is_selected flag).